api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -890,6 +892,43 @@ public static void evictFromSession(Object obj) {
 		getContextDAO().evictFromSession(obj);
 	}
 
+	/**
+	 * Evicts the entity data for a particular entity instance.
+	 *
+	 * @param sessionFactory
+	 * @param entityClass entity class of the instance to evict from the DB cache
+	 * @param object entity instance to evict from the DB cache
+	 */
+	public static void evictSingleEntity(SessionFactory sessionFactory, Class<?> entityClass, OpenmrsObject object) {
```

Comment:
- I think single in this method name is redundant,  **evictEntity** should be enough.
- The class argument is not needed because it can be fetched from the object itself

Reply:
@ibacher is right, the caller should not have to pass the session factory as an argument, this call needs to delegate to the context DAO.